app/src/main/java/org/oppia/app/home/HomeActivityPresenter.kt

```diff
@@ -38,6 +38,7 @@ class HomeActivityPresenter @Inject constructor(private val activity: AppCompatA
       activity.findViewById<View>(R.id.home_activity_drawer_layout) as DrawerLayout,
       toolbar, R.id.nav_home
     )
+    navigationDrawerFragment!!.retainInstance = true
```

Comment:
Actually this is not a correct solution for this.
The problem is in `NavigationDrawerFragment` -> `routeToProfileProgress`, when we do orientation change the activity gets destroyed and the lifecycle of fragment is independent of that and thats why `this.activity!!` is creating issues as it is null. To fix this, we should actually call intent in the `NavigationDrawerFragmentPresenter` to fix this. Like this:


Code in `NavigationDrawerFragmentPresenter`

```
fun openProfileProgress(profileId: Int) {
    activity.startActivity(ProfileProgressActivity.createProfileProgressActivityIntent(activity, profileId))
  }
```

Inside `routeToProfileProgress` you can just call this presenter function.

Reply:
@rt4914 thanks for the suggestion, I have changed the implementation and it is working fine now!